redis/paymentCache.ts

```diff
@@ -32,6 +32,66 @@ export const getPaymentList = async (userId: string): Promise<Payment[]> => {
   return await getCachedPaymentsForUser(userId)
 }
 
+export const getCachedPaymentsForUserWithPagination = async (
+  userId: string,
+  page: number,
+  pageSize: number,
+  orderDesc: boolean
+): Promise<Payment[]> => {
+  const weekKeys = await getCachedWeekKeysForUser(userId)
+
+  const userButtonIds: string[] = (await fetchPaybuttonArrayByUserId(userId))
+    .map(p => p.id)
+
+  const allPayments: Payment[] = []
+  const startIndex = page * pageSize
+  const endIndex = Number(startIndex) + Number(pageSize)
+  let currentIndex = 0
+  const sortedWeekKeys = sortByDate(weekKeys, (orderDesc ? 'descending' : 'ascending'))
+  for (const weekKey of sortedWeekKeys) {
+    const paymentsString = await redis.get(weekKey)
+
+    if (paymentsString === null) {
+      throw new Error(RESPONSE_MESSAGES.CACHED_PAYMENT_NOT_FOUND_404.message)
+    }
+
+    const weekPayments: Payment[] = JSON.parse(paymentsString)
+
+    for (const payment of weekPayments) {
+      if (currentIndex >= startIndex && currentIndex < endIndex) {
+        payment.buttonDisplayDataList = payment.buttonDisplayDataList.filter(d =>
+          userButtonIds.includes(d.id)
+        )
+        allPayments.push(payment)
+      }
+      currentIndex++
+
+      if (currentIndex >= endIndex) {
+        console.log('break')
+        break
+      }
+    }
+    if (currentIndex >= endIndex) {
+      break
+    }
+  }
+  return allPayments
+}
+
+function sortByDate (keys: string[], order: 'ascending' | 'descending'): string[] {
```

Comment:
more clear name: `sortWeekKeysByDate`, since this function is specific for it.

Reply:
No need to do that anymore, since this is obsolete, it can be deleted. However, if you think this could be useful for a future task, just put a comment indicating that and rename it for more clarity.